### PR TITLE
[VM] Enhance vm open-port command

### LINKED ISSUE #1982: [VM] Could the open-port command allow specifying a port?
The simplified `az network nsg rule create` command is awesome, however, it would be great if you could use the `az vm open-port` command as a VM-focused way of achieving the same thing, but without having to look up and manage the NSG explicitly (fewer concept counts for new Azure users would be great!).

Since `az vm create` auto-creates an NSG behind the scenes, being able to expose a port on the VM would help support an easier "getting started" experience with VMs. Instead of the `open-port` command exposing all ports, it would nice to do something like the following:

```shell
az vm create -g <group> -n <name> --image UbuntuLTS
az vm open-port -g <group> -n <name> -p 80
```

Even further, I would actually love if it the `az vm create` command allowed specifying a list of ports to expose on the resulting NSG. Something like the following would be amazing:

```shell
az vm create -g <group> -n <name> --image UbuntuLTS --expose-ports 80,443
```

Both of these help achieve the same goal of preventing a user from needing to think about the NSG until necessary. Since exposing a port is such a common operation, it seems like this would be a nice companion/extension of the already-great abstraction created by the `az vm create` command in the first place.

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #1982.

- **Breaking Change**: Adds required parameter: `--port`. The old default of * doesn't make sense.

### CODE REVIEW

**Comment by derekbekoe** [comment]
Can we make `priority` optional?
If I've just run `az vm create` and then do `az vm open-port`, not sure if I'd know what priority to use.

**Comment by lostintangent** [comment]
@derekbekoe That would be ideal if it was reasonable to allow.

**Comment by derekbekoe** [comment]
Can we include an example of a port range in the help for this param?
Will any of these formats work?
`1000-5000` (all ports from 1000 to 5000 inclusive)
`80,443` (only 80 and 443)

**Comment by tjprescott** [comment]
@derekbekoe the problem with defaulting the priority is that you can't have two inbound rules with the same priority. In essence, it would work the first time and then fail the second. Like `nsg rule create` it will only accept a single port or a range of ports. It will not accept a list (comma separated or otherwise). 

I had an implementation that took a space separated list and simply created multiple rules, but then you immediately run into the problem of priority. Without a good heuristic to go by, I stuck with only creating one rule with each call and making priority required.

**Comment by yugangw-msft** [comment]
> In essence, it would work the first time and then fail the second

@tjprescott, how the error looks like? If the error is good, failing with using the same default priority would be ok, and people can refer to examples section for help.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -155,6 +155,8 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 register_cli_argument('vm open-port', 'vm_name', name_arg_type, help='The name of the virtual machine to open inbound traffic on.')
 register_cli_argument('vm open-port', 'network_security_group_name', options_list=('--nsg-name',), help='The name of the network security group to create if one does not exist. Ignored if an NSG already exists.', validator=validate_nsg_name)
 register_cli_argument('vm open-port', 'apply_to_subnet', help='Allow inbound traffic on the subnet instead of the NIC', action='store_true')
+register_cli_argument('vm open-port', 'port', help="The port or port range to open inbound traffic to. Use '*' to allow traffic to all ports.")
```
Comment (lostintangent) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Could we alias this as `-p` or something else?

**Comment by tjprescott** [comment]
@yugangw-msft the error is pretty clear. I'll leave the priority default to 900 then. Thanks for the feedback. :)

**Pull request merged**